gcloud-java-core/src/main/java/com/google/gcloud/AuthCredentials.java

```diff
@@ -271,9 +315,17 @@ public static ServiceAccountAuthCredentials createFor(String account, PrivateKey
    */
   public static ServiceAccountAuthCredentials createForJson(InputStream jsonCredentialStream)
       throws IOException {
-    GoogleCredential tempCredentials = GoogleCredential.fromStream(jsonCredentialStream);
-    return new ServiceAccountAuthCredentials(tempCredentials.getServiceAccountId(),
-        tempCredentials.getServiceAccountPrivateKey());
+    GoogleCredentials tempCredentials = GoogleCredentials.fromStream(jsonCredentialStream);
```

Comment:
It would be nicer to change this method to return AuthCredentials and to also support `UserCredentials(String clientId, String clientSecret, String refreshToken) `.